### PR TITLE
integ tests: Add bond with dhcp and slave

### PR DESCRIPTION
**Pull request by SvetaHBar**

### CODE REVIEW

**Comment by tyll** [comment]
One line is too long which makes Travis fail.

**Comment by tyll** [comment]
ci test please

File: tests/integration/dhcp_test.py [hunk 1]
```diff
@@ -148,6 +163,43 @@ def test_dhcp_with_addresses(dhcp_env):
...
+        INTERFACES: [
+            {
+                'name': 'bond99',
+                'type': 'bond',
+                'state': 'up',
+                'ipv4': {
+                    'enabled': True,
+                    'dhcp': True,
+                    'address': [
+                        {'ip': '192.168.122.250', 'prefix-length': 24}
```
Comment (tyll) [inline tests/integration/dhcp_test.py]: Please use rfc5737 addresses here and even better use a constant as you do in https://github.com/nmstate/nmstate/blob/master/tests/integration/static_ip_address_test.py#L26

Comment (SvetaHBar) [reply tests/integration/dhcp_test.py]: Done

**Comment by EdDev** [comment]
Based on the logs, this PR is stuck on https://nmstate.atlassian.net/browse/NMSTATE-71.
It shows that starting the master (bond), does not autoconnect the slaves, causing the dhcp client to timeout (as the slaves are not brought up).

**Comment by EdDev** [comment]
@tyll could you please revisit this PR?

File: tests/integration/dhcp_test.py [hunk 2]
```diff
@@ -148,6 +163,43 @@ def test_dhcp_with_addresses(dhcp_env):
...
+                    'slave': [DHCP_CLI_NIC],
+                    'options':
+                        {'miimon': '140'}
+                }
+
+            },
+            {
+                'name': DHCP_CLI_NIC,
+                'type': 'ethernet',
+                'state': 'up'}
```
Comment (EdDev) [inline tests/integration/dhcp_test.py]: The closing parenthesis needs to appear in the next line.

File: tests/integration/dhcp_test.py [hunk 3]
```diff
@@ -148,6 +163,43 @@ def test_dhcp_with_addresses(dhcp_env):
...
+    desired_state = {
+        INTERFACES: [
+            {
+                'name': 'bond99',
+                'type': 'bond',
+                'state': 'up',
+                'ipv4': {
+                    'enabled': True,
+                    'dhcp': True,
+                    'address': [
```
Comment (EdDev) [inline tests/integration/dhcp_test.py]: nmstate ignores this static IP, I would expect the test to fail.

File: tests/integration/dhcp_test.py (hunk 2 above)
Comment (SvetaHBar) [reply tests/integration/dhcp_test.py]: done

**Comment by tyll** [comment]
ci test please

**Comment by EdDev** [comment]
I do not see the errors on the CI on my local machine.
I only see that the test fail on the assertion due to the static IP (which collides with the `dhcp=True`.

We need to dig into the NM logs (journal).

**Comment by tyll** [comment]
@EdDev the stdci shows run timeouts: https://jenkins.ovirt.org/blue/organizations/jenkins/nmstate_nmstate_standard-check-pr/detail/nmstate_nmstate_standard-check-pr/899/pipeline #243

**Comment by EdDev** [comment]
@tyll The problem is here: https://jenkins.ovirt.org/blue/organizations/jenkins/nmstate_nmstate_standard-check-pr/detail/nmstate_nmstate_standard-check-pr/899/pipeline/#step-190-log-132

The retry does not help in this case. So specifically on the CI run, with this setup scenario, it does not work.

Any chance you can ask someone from the NM team to have a look?

File: tests/integration/dhcp_test.py [hunk 4]
```diff
@@ -148,6 +163,44 @@ def test_dhcp_with_addresses(dhcp_env):
...
+                        {'ip': IPV4_ADDRESS1, 'prefix-length': 24}
+                    ]
+                },
+
+                'link-aggregation': {
+                    'mode': 'balance-rr',
+                    'slave': [DHCP_CLI_NIC],
```
Comment (EdDev) [inline tests/integration/dhcp_test.py]: It should be `slaves`

File: tests/integration/dhcp_test.py [hunk 5]
```diff
@@ -148,6 +163,44 @@ def test_dhcp_with_addresses(dhcp_env):
...
+                    'options':
+                        {'miimon': '140'}
+                }
+
+            },
+            {
```
Comment (EdDev) [inline tests/integration/dhcp_test.py]: You do not need to declare the slave.

File: tests/integration/dhcp_test.py (hunk 4 above)
Comment (SvetaHBar) [reply tests/integration/dhcp_test.py]: Fixed

File: tests/integration/dhcp_test.py (hunk 5 above)
Comment (SvetaHBar) [reply tests/integration/dhcp_test.py]: Done

**Pull request merged**